src/Distributed/multi_architectures.jl

```diff
@@ -140,17 +147,21 @@ end
 ##### Rank connectivity graph
 #####
 
-struct RankConnectivity{E, W, N, S, T, B}
-      east :: E
-      west :: W
-     north :: N
-     south :: S
-       top :: T
-    bottom :: B
+struct RankConnectivity{E, W, N, S, T, B, SW, SE, NW, NE}
+         east :: E
+         west :: W
+        north :: N
+        south :: S
+          top :: T
+       bottom :: B
```

Comment:
do we split the domain in the vertical? why do we need top/bottom connectivity?

Reply:
we cannot split in the vertical, that is legacy code. I could clean it all up here, or in a another PR. I am ok with both.

Reply:
isn't this code added by this PR? how is it legacy?

Reply:
just changed the spacing for aesthetic purposes, this is in main
https://github.com/CliMA/Oceananigans.jl/blob/19dac0b8f6b621057e250ecf0510b3c4c91915e9/src/Distributed/multi_architectures.jl#L148-L149